src/app/ReadHandler.h

```diff
@@ -172,14 +174,17 @@ class ReadHandler : public Messaging::ExchangeDelegate
 
     /*
      * Set the reporting intervals for the subscription. This SHALL only be called
-     * from the OnSubscriptionRequested callback above.
+     * from the OnSubscriptionRequested callback above. The restriction is as below
+     * MinIntervalFloor ≤ MaxInterval ≤ MAX(SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT, MaxIntervalCeiling)
+     * Where SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT is set to 60m in the spec.
      */
-    CHIP_ERROR SetReportingIntervals(uint16_t aMinInterval, uint16_t aMaxInterval)
+    CHIP_ERROR SetReportingIntervals(uint16_t aMaxInterval)
     {
         VerifyOrReturnError(IsIdle(), CHIP_ERROR_INCORRECT_STATE);
-        VerifyOrReturnError(aMinInterval <= aMaxInterval, CHIP_ERROR_INVALID_ARGUMENT);
-
-        mMinIntervalFloorSeconds   = aMinInterval;
+        VerifyOrReturnError(mMinIntervalFloorSeconds <= aMaxInterval, CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrReturnError(aMaxInterval <=
+                                std::max(kSubscriptionMaxIntervalPublisherLimit, mMaxIntervalCeilingSeconds),
+                            CHIP_ERROR_INVALID_ARGUMENT);
         mMaxIntervalCeilingSeconds = aMaxInterval;
```

Comment:
We should re-name this member variable to `mMaxInterval`, since it logically speaking, is tracking the final negotiated max interval. It happens to be initialized to the value set in the request, but that's just an initial value.

Otherwise, this becomes confusing, and won't match the spec.

Reply:
done, thanks